cvat-sdk/cvat_sdk/auto_annotation/driver.py

```diff
@@ -0,0 +1,226 @@
+                        raise BadFunctionError("function output shape element with preset source")
+                    element.source = "auto"
+
+                    if element.frame != 0:
+                        raise BadFunctionError(
+                            f"function output shape element with unexpected frame number ({element.frame})"
+                        )
+
+                    element.frame = ds_frame
+
+                    if element.type.value != "points":
+                        raise BadFunctionError(
+                            f"function output skeleton with element type other than 'points' ({element.type.value})"
+                        )
+
+                    try:
+                        mapped_sl_id = mapped_label.sublabel_mapping[element.label_id]
+                    except KeyError:
+                        raise BadFunctionError(
+                            f"function output shape with unknown sublabel ID ({element.label_id})"
+                        )
+
+                    if not mapped_sl_id:
+                        continue
+
+                    if mapped_sl_id in seen_sl_ids:
+                        raise BadFunctionError(
+                            "function output skeleton with multiple elements with same sublabel"
+                        )
+
+                    element.label_id = mapped_sl_id
+
+                    seen_sl_ids.add(mapped_sl_id)
+
+                    new_elements.append(element)
+
+                if len(new_elements) != mapped_label.expected_num_elements:
+                    # new_elements could only be shorter than expected,
+                    # because the reverse would imply that there are more distinct sublabel IDs
+                    # than are actually defined in the dataset.
+                    assert len(new_elements) < mapped_label.expected_num_elements
+
+                    raise BadFunctionError(
+                        f"function output skeleton with fewer elements than expected ({len(new_elements)} vs {mapped_label.expected_num_elements})"
+                    )
+
+                shape.elements[:] = new_elements
+            else:
+                if getattr(shape, "elements", None):
+                    raise BadFunctionError("function output non-skeleton shape with elements")
+
+        shapes[:] = new_shapes
+
+
+def annotate_task(client: Client, task_id: int, function: DetectionFunction) -> None:
+    dataset = TaskDataset(client, task_id)
+
+    assert isinstance(function.spec, DetectionFunctionSpec)
+
+    mapper = _AnnotationMapper(client.logger, function.spec.labels, dataset.labels)
+
+    shapes = []
+
+    context = DetectionFunctionContext()
+
+    for sample in dataset.samples:
+        frame_shapes = function.detect(context, sample.media.load_image())
+        mapper.validate_and_remap(frame_shapes, sample.frame_index)
+        shapes.extend(frame_shapes)
+
+    client.logger.info("Uploading annotations to task %d", task_id)
```

Comment:
Here we force users to wait till all frames are annotated. In CLI we should be able to show progress and also I believe it will be useful to have a configuration option to upload data in batches. What do you think?

Reply:
Regarding progress: yes we should.

Regarding uploading data in batches: I think notionally it could be useful, but in practice there are problems:

* I'm not sure if the API supports partial uploads like this.
* If the annotation fails mid-way, there's no way to restart it from where it left off (and it it doesn't fail mid-way, then what's the point in uploading in batches?).

Reply:
I added progress reporting.